src/config.rs

```diff
@@ -260,6 +398,72 @@ pub struct ArbConfig {
     pub finality: u8,
 }
 
+#[serde_as]
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(crate = "serde_crate")]
+#[serde(bound(serialize = "T: Display"))]
+#[serde(bound(deserialize = "T: FromStr, T::Err: Display"))]
+pub struct TradeableAmounts<T>
+where
+    T: FromStr + Display,
+    T::Err: Display,
+{
+    /// If specified, the minimum acceptable amount to trade (>=)
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    #[serde(default)]
+    pub min_amount: Option<T>,
+    /// If specified, the maximum acceptable amount to trade (<=)
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    #[serde(default)]
+    pub max_amount: Option<T>,
+}
+
+impl<T> TradeableAmounts<T>
+where
+    T: FromStr + Display,
+    T::Err: Display,
+{
+    /// Map tradeable amounts into new type
+    pub fn map<U>(self) -> TradeableAmounts<U>
+    where
+        T: Into<U>,
+        U: FromStr + Display,
+        U::Err: Display,
+    {
+        TradeableAmounts {
+            min_amount: self.min_amount.map(|min| min.into()),
+            max_amount: self.max_amount.map(|max| max.into()),
+        }
+    }
+
+    /// Validate a given amount based on potential rules (min, max)
+    pub fn validate_amount(&self, amount: T) -> Result<(), Error>
+    where
+        T: PartialOrd<T> + Copy,
+    {
+        use config::ConfigError::Message;
```

Comment:
Why is this imported here and nowhere else?

Reply:
It was imported at two places and not used everywhere. I import them at function scope level to shorten the error return lines, as this is used elsewhere it's now imported everywhere with c003b47c433a812edb3b23359e843a9555e39bb3